x-pack/plugins/security_solution/server/utils/runtime_types.ts

```diff
@@ -0,0 +1,126 @@
+import * as rt from 'io-ts';
+import { failure } from 'io-ts/lib/PathReporter';
+import get from 'lodash/get';
+
+type ErrorFactory = (message: string) => Error;
+
+export type GenericIntersectionC =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | rt.IntersectionC<[any, any]>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | rt.IntersectionC<[any, any, any]>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | rt.IntersectionC<[any, any, any, any]>
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  | rt.IntersectionC<[any, any, any, any, any]>;
+
+export const createPlainError = (message: string) => new Error(message);
+
+export const throwErrors = (createError: ErrorFactory) => (errors: rt.Errors) => {
+  throw createError(failure(errors).join('\n'));
+};
+
+export const decodeOrThrow = <A, O, I>(
+  runtimeType: rt.Type<A, O, I>,
+  createError: ErrorFactory = createPlainError
+) => (inputValue: I) =>
+  pipe(runtimeType.decode(inputValue), fold(throwErrors(createError), identity));
+
+const getProps = (
+  codec:
+    | rt.HasProps
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    | rt.RecordC<rt.StringC, any>
+    | GenericIntersectionC
+): rt.Props | null => {
+  if (codec == null) {
+    return null;
+  }
+  switch (codec._tag) {
+    case 'DictionaryType':
+      if (codec.codomain.props != null) {
+        return codec.codomain.props;
+      }
+      const dTypes: rt.HasProps[] = codec.codomain.types;
+      return dTypes.reduce<rt.Props>((props, type) => Object.assign(props, getProps(type)), {});
+    case 'RefinementType':
+    case 'ReadonlyType':
+      return getProps(codec.type);
+    case 'InterfaceType':
+    case 'StrictType':
+    case 'PartialType':
+      return codec.props;
+    case 'IntersectionType':
+      const iTypes = codec.types as rt.HasProps[];
+      return iTypes.reduce<rt.Props>((props, type) => {
+        return Object.assign(props, getProps(type) as rt.Props);
+      }, {} as rt.Props) as rt.Props;
+    default:
+      return null;
+  }
+};
+
+const getExcessProps = (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  props: rt.Props | rt.RecordC<rt.StringC, any>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  r: any
+): string[] => {
+  return Object.keys(r).reduce<string[]>((acc, k) => {
+    const codecChildren = get(props, [k]);
+    const childrenProps = getProps(codecChildren);
+    const childrenObject = r[k] as Record<string, unknown>;
+    if (codecChildren != null && childrenProps != null && codecChildren._tag === 'DictionaryType') {
+      const keys = Object.keys(childrenObject);
+      return [
+        ...acc,
+        ...keys.reduce<string[]>(
+          (kAcc, i) => [...kAcc, ...getExcessProps(childrenProps, childrenObject[i])],
+          []
+        ),
+      ];
+    }
+    if (get(props, [k]) != null && childrenProps != null) {
```

Comment:
you can replace `get(props, [k])` with `codecChildren` defined above https://github.com/elastic/kibana/pull/76472/files#diff-e17d23200e77f18f173d4520b3fe0440R79